### ISSUE TITLE
Lint/Syntax: unexpected token kDO

### ISSUE THREAD

**Issue by GBH** [bbatsov/rubocop#5573]
## Expected behavior

Suddenly I started to get rubocop errors on your regular `assert_difference` helper methods. Seems that https://github.com/whitequark/parser got bumped to 2.5.0.0 few days ago and it changed something.

## Actual behavior

```
test/controllers/comfy/admin/cms/files_controller_test.rb:224:58: E: Lint/Syntax: unexpected token kDO
(Using Ruby 2.2 parser; configure using TargetRubyVersion parameter, under AllCops)
    assert_difference [file_count, attachment_count], -1 do
```

Putting brackets around makes it go away:

```ruby
assert_difference("User.count", 1) do
  # something
end
```

Edit. Seems it's the arguments issue:

```ruby 
function foo do #ok
function foo, bar do #not ok
```

## RuboCop version

```
$ rubocop -V
0.52.1 (using Parser 2.5.0.0, running on ruby 2.5.0 x86_64-linux)
```

**Comment by pocke** [comment]
It is not a RuboCop's bug. Can you open an issue on whitequark/parser?




Reproducing
===


```ruby
# test.rb
assert_difference [file_count, attachment_count], -1 do
end
```

```bash
$ ruby -vc test.rb
ruby 2.5.0p0 (2017-12-25 revision 61468) [x86_64-linux]
Syntax OK

$ ruby-parse _2.5.0.0_ test.rb
test.rb:1:54: error: unexpected token kDO
test.rb:1: assert_difference [file_count, attachment_count], -1 do
test.rb:1:                                                      ^~

$ ruby-parse _2.4.0.1_ test.rb
(block
  (send nil :assert_difference
    (array
      (send nil :file_count)
      (send nil :attachment_count))
    (int -1))
  (args) nil)
```

**Comment by GBH** [comment]
Posted there: https://github.com/whitequark/parser/issues/447
